src/operations/write.ts

```diff
@@ -102,7 +102,14 @@ export const writeOptimistic = (
           const fieldSelect = getSelectionSet(operation);
```

Comment:
I think `fieldSelect[0]` is just the first field 😅  but I might've spotted my mistake, should it be

```suggestion
          const fieldSelect = getSelectionSet(node);
```

^this may be what I was meaning to write 😅